### PR TITLE
Fixed: O(n) seek in buffered column. Benchmarks great again.

### PR DESCRIPTION
**Pull request by MBkkt**

### CODE REVIEW

File: tests/index/buffered_column_test.cpp [hunk 1]
```diff
@@ -178,12 +178,12 @@ void AssertIteratorCornerCases(const irs::BufferedColumn& column,
...
-    ASSERT_FALSE(irs::doc_limits::valid(it.seek(irs::doc_limits::invalid())));
+    const auto value = it.seek(irs::doc_limits::invalid());
+    ASSERT_EQ(value, it.value());
     if (!expected_values.empty()) {
-      ASSERT_TRUE(it.next());
-      ASSERT_EQ(1, it.value());
+      ASSERT_EQ(value, 1);
     } else {
-      ASSERT_FALSE(it.next());
+      ASSERT_EQ(value, irs::doc_limits::eof());
```
Comment (Dronplane) [inline tests/index/buffered_column_test.cpp:186]: Why dropped check for `next` after eof? This is valid and well defined 
// Once iterator is exhausted:
//   - `next()` must constantly return `false`
//   - `seek()`, `shallow_seek()` to any value must return `doc_limits::eof()`
//   - `value()` must return `doc_limits::eof()`

**Review by Dronplane — COMMENTED** [review]
LGTM. See note about test

File: tests/index/buffered_column_test.cpp (hunk 1 above)
Comment (MBkkt) [reply tests/index/buffered_column_test.cpp:186]: This test already has such check

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**